src/common/transformations/src/transformations/common_optimizations/align_mixed_fp32_fp16_types.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/align_mixed_fp32_fp16_types.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/precision_sensitive_attribute.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "transformations/convert_precision.hpp"
+#include "transformations/rt_info/decompression.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+
+using namespace ov;
+
+bool ov::pass::AlignMixedFP32FP16Types::run_on_model(const std::shared_ptr<ov::Model>& model) {
+    RUN_ON_MODEL_SCOPE(AlignMixedFP32FP16Types);
+
+    std::function<bool(const std::shared_ptr<Node>&)> insert_converts_before = [&](const std::shared_ptr<Node>& node) {
+        bool is_changed = false;
+        for (const auto& input : node->inputs()) {
+            const auto& incoming_output = input.get_source_output();
+            const auto& incoming_node = incoming_output.get_node_shared_ptr();
+
+            if (fp16_compression_is_disabled(incoming_node))
+                continue;  // we are in the middle
+
+            if (!incoming_output.get_element_type().is_real())
+                continue;
+
+            const auto& existing_convert = std::dynamic_pointer_cast<opset10::Convert>(incoming_node);
+            if (existing_convert) {
+                existing_convert->set_convert_element_type(element::f32);
```

Comment:
`existing_convert->set_convert_element_type(element::f32);` is risky change because it can have other consumers and you can create type inconsistency for them

Reply:
Agree, if `Convert` has several destinations this lines can cause problems. I removed it's, since it's redundant.
Any additional Converts like Convert_to_f32->Convert_to_f32 will be removed right away here
https://github.com/openvinotoolkit/openvino/blob/988847f559501c63c486f0e081fdafa1cba8c88b/src/common/transformations/src/transformations/convert_precision.cpp#L219-L227